File: server/internal/model/input/compliancein/order.go
package compliancein

// OrderCreateInp 创建服务订单入参
type OrderCreateInp struct {
	MemberId    uint64
	PlanId      uint64
	DiagnosisId uint64
}

// OrderCreateModel 创建订单出参
type OrderCreateModel struct {
	OrderId uint64  `json:"orderId"`
	Status  string  `json:"status"`
	Amount  float64 `json:"amount"`
	PlanId  uint64  `json:"planId"`
}

// ConsentInp 合规同意留痕入参
type ConsentInp struct {
	MemberId           uint64
	OrderId            uint64
	Type               string
	DocumentVersion    string
	Acknowledgments    *RiskAcknowledgments
	PlanConfirmed      bool
	Ip                 string
	UserAgent          string
}

// RiskAcknowledgments 风险告知四条确认
type RiskAcknowledgments struct {
	ComplianceNotEvasion bool `json:"complianceNotEvasion"`
	NoAuditGuarantee     bool `json:"noAuditGuarantee"`
	NoFakeInvoice        bool `json:"noFakeInvoice"`
	TaxDependsOnReality  bool `json:"taxDependsOnReality"`
}

// ConsentModel 同意留痕出参
type ConsentModel struct {
	ConsentId uint64 `json:"consentId"`
	Type      string `json:"type"`
	OrderId   uint64 `json:"orderId"`
}

// SignInp 电子签约入参
type SignInp struct {
	MemberId    uint64
	OrderId     uint64
	LegalName   string
	Ip          string
	UserAgent   string
}

// SignModel 签约出参
type SignModel struct {
	OrderId   uint64 `json:"orderId"`
	Status    string `json:"status"`
	SignedAt  string `json:"signedAt"`
	OpcId     uint64 `json:"opcId,omitempty"`
	OpcStatus string `json:"opcStatus,omitempty"`
}

// ActiveOrderModel 当前活跃订单
type ActiveOrderModel struct {
	OrderId     uint64  `json:"orderId"`
	PlanId      uint64  `json:"planId"`
	PlanName    string  `json:"planName,omitempty"`
	PlanTier    string  `json:"planTier,omitempty"`
	DiagnosisId uint64  `json:"diagnosisId,omitempty"`
	Status      string  `json:"status"`
	Amount      float64 `json:"amount,omitempty"`
	SignedAt    string  `json:"signedAt,omitempty"`
	LegalName   string  `json:"legalName,omitempty"`
	CreatedAt   string  `json:"createdAt,omitempty"`
}

File: web/src/api/frontend/compliance/order.ts
/**
 * 前台合规签约 API（需登录）
 * @module api/frontend/compliance/order
 */
import { memberRequest } from '@/utils/http'
import type { ServicePlan } from './diagnosis'

/** 订单状态 */
export type OrderStatus = 'pending' | 'active' | 'cancelled'

/** 合规订单 */
export interface ComplianceOrder {
  orderId: number | string
  /** @deprecated 兼容旧字段 */
  id?: number | string
  planId: number | string
  planTier?: 'basic' | 'advanced' | 'premium'
  planName?: string
  diagnosisId?: number | string
  status: OrderStatus
  amount?: number
  signedAt?: string
  createdAt?: string
}

/** 创建订单参数 */
export interface CreateOrderParams {
  planId?: number | string
  planTier?: string
  diagnosisId?: number | string
}

/** 风险告知四条确认（对齐后端 RiskAcknowledgments） */
export interface RiskAcknowledgments {
  complianceNotEvasion: boolean
  noAuditGuarantee: boolean
  noFakeInvoice: boolean
  taxDependsOnReality: boolean
}

/** 风险告知 / 方案确认参数 */
export interface ConsentParams {
  orderId: number | string
  type: 'risk_disclosure' | 'plan_confirm'
  acknowledgments?: RiskAcknowledgments
  planConfirmed?: boolean
  documentVersion?: string
}

/** 电子签约参数 */
export interface SignParams {
  orderId: number | string
  legalName: string
}

/** 签约结果 */
export interface SignResult {
  orderId: number | string
  opcId?: number | string
  signedAt: string
}

/** 创建合规订单 */
export function createComplianceOrder(params: CreateOrderParams) {
  return memberRequest.post<ComplianceOrder>({
    url: '/compliance/order',
    data: params
  })
}

/** 提交风险告知 / 方案确认 */
export function submitConsent(params: ConsentParams) {
  return memberRequest.post<void>({
    url: '/compliance/consent',
    data: params
  })
}

/** 完成电子签约 */
export function signAgreement(params: SignParams) {
  return memberRequest.post<SignResult>({
    url: '/compliance/sign',
    data: params
  })
}

/** 获取当前活跃订单（若无有效订单则返回 null） */
export async function getActiveOrder(): Promise<ComplianceOrder | null> {
  const res = await memberRequest.get<ComplianceOrder | null>({
    url: '/compliance/order/active'
  })
  const orderId = Number(res?.orderId ?? res?.id ?? 0)
  if (!orderId || !res?.status || res.status === 'cancelled') {
    return null
  }
  return res
}

export type { ServicePlan }
